Framework/Algorithms/src/ReflectometryReductionOne.cpp

```diff
@@ -612,7 +630,38 @@ void ReflectometryReductionOne::exec() {
   }
 
   IvsQ = this->toIvsQ(IvsLam, correctDetectorPositions, theta, isPointDetector);
-
+  double momentumTransferMinimum = getProperty("MomentumTransferMinimum");
+  double momentumTransferStep = getProperty("MomentumTransferStep");
+  double momentumTransferMaximum = getProperty("MomentumTransferMaximum");
+  if (!isDefault("MomentumTransferMinimum") &&
+      !isDefault("MomentumTransferStep") &&
+      !isDefault("MomentumTransferMaximum")) {
+    MantidVec QParams;
+    QParams.push_back(momentumTransferMinimum);
+    QParams.push_back(-momentumTransferStep);
+    QParams.push_back(momentumTransferMaximum);
+    IAlgorithm_sptr algRebin = this->createChildAlgorithm("Rebin");
+    algRebin->initialize();
+    algRebin->setProperty("InputWorkspace", IvsQ);
+    algRebin->setProperty("OutputWorkspace", IvsQ);
+    algRebin->setProperty("Params", QParams);
+    algRebin->execute();
+    if (!algRebin->isExecuted())
+      throw std::runtime_error("Failed to run Rebin algorithm");
+    IvsQ = algRebin->getProperty("OutputWorkspace");
+  }
```

Comment:
If `MomentumTransferMinimum`, `MomentumTransferStep` and `MomentumTransferMaximum` are not set, we should try to calculate them. To do so, we would need to apply the formula:

```
Q = 4 * pi / lambda * sin (theta)
```
to get the minimum and maximum values, and run `CalculateResolution` to get the step. But this is only possible if `ThetaIn` has been supplied by the user. If `ThetaIn` has not been set, we will have to wait until `SpecularReflectionCalculateTheta` is run, which happens in `ReflectometryReductionOne`.